packages/scheduler/lib/scheduler.ts

```diff
@@ -0,0 +1,198 @@
+import { isMainThread } from 'node:worker_threads';
+import type { JsonObject, SchedulingProps, Task, TaskState } from './types';
+import * as tasks from './models/tasks.js';
+import type { Result } from '@nangohq/utils';
+import { stringifyError, getLogger } from '@nangohq/utils';
+import { MonitorWorker } from './monitor.worker.js';
+
+const logger = getLogger('Scheduler');
+
+export class Scheduler {
+    // TODO: scheduling recurring tasks
+
+    private monitor: MonitorWorker | null = null;
+    private onCallbacks: Record<TaskState, (task: Task) => void>;
+
+    /**
+     * Scheduler
+     * @constructor
+     * @param on - Callbacks for task state transitions
+     * @returns Scheduler
```

Comment:
I would be very cautious about jsdoc, unless we add an eslint package to enforce their consistency it's not going to be maintained properly

Reply:
It is a risk indeed. Would you prefer no jsdoc?

Reply:
I am keeping the comments for now.